### PR TITLE
cmd/snap-update-ns: handle mountpoint removal failures with EBUSY

### PR DESCRIPTION
**Pull request by mardy**
In the field it has been seen that this is an error which can happen,
probably due to some mistakes on how writable mimics are built, or the
ordering in which they are mounted.

This is a workaround, the issue is still being investigated.

Based on top of https://github.com/snapcore/snapd/pull/11920

### CODE REVIEW

**Comment by pedronis** [comment]
@mardy could you add to description an example of mount entry that we fail to unmount ? also do I undestand we get an EBUSY out of a --lazy umount?

File: cmd/snap-update-ns/change.go [hunk 1]
```diff
@@ -465,6 +465,18 @@ func (c *Change) lowLevelPerform(as *Assumptions) error {
 				logger.Debugf("cannot remove a mount point on read-only filesystem %q", path)
 				return nil
 			}
+			if err == syscall.EBUSY {
+				// It's still unclear how this can happen. For the time being
+				// let the operation succeed and log the event.
+				logger.Noticef("cannot remove mount point, got EBUSY: %q", path)
+				if isMount, err := osutil.IsMounted(path); isMount {
+					mounts, _ := osutil.LoadMountInfo()
+					logger.Noticef("%q is still mounted:\n%s", path, mounts)
```
[the remark this answers is not in the document]
Comment (mardy) [reply cmd/snap-update-ns/change.go]: Updated.

**Comment by mvo5** [comment]
I got access to the logs from a system that works with this PR (i.e. current stable will fail, stable+this-pr will work). But AFAICT there is no "cannot remove mount point, got EBUSY:" in the logs.

**Comment by pedronis** [comment]
> I got access to the logs from a system that works with this PR (i.e. current stable will fail, stable+this-pr will work). But AFAICT there is no "cannot remove mount point, got EBUSY:" in the logs.

either it means that the bug doesn't always happen?  or that the "unmountOrder := unsortedCurrent" is enough as a fix?

**Comment by mardy** [comment]
> > I got access to the logs from a system that works with this PR (i.e. current stable will fail, stable+this-pr will work). But AFAICT there is no "cannot remove mount point, got EBUSY:" in the logs.
> 
> either it means that the bug doesn't always happen? or that the "unmountOrder := unsortedCurrent" is enough as a fix?

No, that change is not enough to fix the issue. The thing is that all the output from `snap-update-ns` is discarded unless the command fails. I can change that, if you agree.

**Pull request merged**